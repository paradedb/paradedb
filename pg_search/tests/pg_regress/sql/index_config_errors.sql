-- test that when index config isn't passed correctly, the meaningful errors are returned

\echo 'Test: Index configuration errors'

DROP TABLE IF EXISTS test_index_config_errors CASCADE;
CREATE TABLE test_index_config_errors
(
    id    serial8 not null primary key,
    name  text
);

CREATE INDEX idx_chunks_bm25 ON test_index_config_errors
    USING bm25 (id, name)
    WITH (
    key_field = 'id',
    text_fields ='{
        "some_wrong_key": {"tokenizer": {"type": "default"}}
    }'
    );


CREATE INDEX idx_chunks_bm25 ON test_index_config_errors
    USING bm25 (id, name)
    WITH (
    key_field = 'id',
    text_fields ='{
        "name": {"tokenizer": {"type": "some_wrong_type"}}
    }'
    );


CREATE INDEX idx_chunks_bm25 ON test_index_config_errors
    USING bm25 (id, name)
    WITH (
    key_field = 'id',
    text_fields ='{
        "id": {"tokenizer": {"type": "default"}}
    }'
    );



CREATE INDEX idx_chunks_bm25 ON test_index_config_errors USING bm25 (id, name);
CREATE INDEX idx_chunks_bm25 ON test_index_config_errors USING bm25 (id, name) WITH (text_fields ='{"id": {"tokenizer": {"type": "default"}}}');


DROP TABLE test_index_config_errors CASCADE;
