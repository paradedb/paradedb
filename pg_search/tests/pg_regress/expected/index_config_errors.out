-- test that when index config isn't passed correctly, the meaningful errors are returned
\echo 'Test: Index configuration errors'
Test: Index configuration errors
DROP TABLE IF EXISTS test_index_config_errors CASCADE;
CREATE TABLE test_index_config_errors
(
    id    serial8 not null primary key,
    name  text
);
CREATE INDEX idx_chunks_bm25 ON test_index_config_errors
    USING bm25 (id, name)
    WITH (
    key_field = 'id',
    text_fields ='{
        "some_wrong_key": {"tokenizer": {"type": "default"}}
    }'
    );
ERROR:  field type should have been set for `some_wrong_key`
CREATE INDEX idx_chunks_bm25 ON test_index_config_errors
    USING bm25 (id, name)
    WITH (
    key_field = 'id',
    text_fields ='{
        "name": {"tokenizer": {"type": "some_wrong_type"}}
    }'
    );
ERROR:  field config should be valid for SearchFieldConfig::name: unknown tokenizer type: some_wrong_type
DROP TABLE test_index_config_errors CASCADE;
